dep/checkRE2.chpl

```diff
@@ -0,0 +1,14 @@
+use Version;
+
+proc main() {
+  if chplVersion < createVersion(1,25) {
+    use Regexp;
+    var myRegex = compile("a+");
+  }
+  else {
+    use Regex;
+    var myRegex = compile("a+");
+  }
+  writeln("Chapel correctly made with RE2 support\n");

```

Comment:
Noted on gitter but capturing here:  Unfortunately, this doesn't work as intended because the Chapel compiler today parses all `use` and `import` statements regardless of the scope they appear in.  So this will attempt to parse both `Regexp` and `Regex` which works for Chapel 1.25.x because both modules exist there (`Regexp` as a deprecation placeholder that will be removed soon), but will not work for Chapel 1.24.x because `Regex` didn't exist there yet.

Building Arkouda itself does work because of `src/compat/lt-125/ArkoudaRegexCompat.chpl` which handles this in a different way.